File: 10275.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
	int t; cin >> t;

	for (int i = 0; i < t; i++) {
		int n;
		long long a, b; cin >> n >> a >> b;
		long long gold = pow(2, n);

		int day = 0;
		long long a_gold = 0, b_gold = gold;
		while (1) {
			day++;
			long long half = b_gold / 2;
			if (a_gold + half <= a) a_gold += half;
			b_gold /= 2;

			if (a_gold == a) break;
		}

		cout << day << endl;
	}
}

File: 10275.py
for i in range(int(input())):
    n, a, b = map(int, input().split())
    gold = 2 ** n

    day = 0
    a_gold = 0; b_gold = gold
    while 1:
        day += 1
        half = b_gold // 2
        if a_gold + half <= a:
            a_gold += half
        b_gold //= 2

        if a_gold == a: break

    print(day)
